src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -369,23 +369,40 @@ private SelectItem GenerateExpandItem(ExpandTermToken tokenIn)
                 currentNavProp = ParseComplexTypesBeforeNavigation(currentComplexProp, ref firstNonTypeToken, pathSoFar);
             }
 
+            bool isRef = false;
+            bool isCount = false;
+            bool hasDerivedType = false;

```

Comment:
`hasTypeSegment` better maybe?

Reply:
Used `hasDerivedTypeSegment` since there are other instances where we have TypeSegments before the first non-type segment e.g `Fully.Qualified.Namespace.Employee/Friends`